src/Compilers/CSharp/Portable/CodeGen/EmitStackAllocInitializer.cs

```diff
@@ -47,7 +47,7 @@ private void EmitStackAllocInitializers(TypeSymbol type, BoundArrayInitializatio
                         EmitElementStackAllocInitializers(elementType, initExprs, includeConstants: false);
                     }
                 }
-                else if (elementType.SpecialType.SizeInBytes() == 1)
+                else if (elementType.EnumUnderlyingTypeOrSelf().SpecialType.SizeInBytes() == 1)
```

Comment:
Not related to this PR: we could also lift the 1 byte restriction as was done in the [CreateSpan PR](https://github.com/dotnet/roslyn/pull/61414). Somehow I didn't realize until now.

Reply:
https://github.com/dotnet/roslyn/pull/61414 doesn't do anything on stackalloc, unlike https://github.com/dotnet/roslyn/pull/57123 but I think that's not merged in.

Reply:
Right, I mean we can do something like https://github.com/dotnet/roslyn/pull/61414 but for stackalloc. I didn't realize we'd already done a demo branch with exactly that. Thanks for digging up https://github.com/dotnet/roslyn/pull/57123. I'll ping there to see whether we should make a real PR with that change.

Reply:
@jcouv, what became of this? Even just emitting it as a copy from the data section (which the JIT should be able to unroll in many cases) would likely be useful.

Reply:
@stephentoub It looks like the hackathon project didn't transfer into a committed project/PR. The [last comment from Jared there](https://github.com/dotnet/roslyn/pull/57123#issuecomment-1108741115) is that this just needs keyboard time, but we support it in principle. 
We're out of dev bandwidth for .NET 8 (ie. at least for one month) unless urgent. Let's discuss around that time. We could file an 17.9 issue (refresh, resolve conflicts and bring https://github.com/dotnet/roslyn/pull/57123 into `main`) for tracking purpose.

Reply:
I opened https://github.com/dotnet/roslyn/issues/69325. Thanks.